### PR TITLE
Allowing a suffix at the end of filename in batch upload imageuploader, like is done in imaging uploader now

### PR DESCRIPTION
**Pull request by MounaSafiHarab**
This is to be in line with:
https://github.com/aces/Loris/pull/2100
in the Loris 16.1 release

Usage details are here:
https://github.com/aces/Loris-MRI/pull/133

### CODE REVIEW

**Comment by gluneau** [comment]
Can you add a usage section?

```
./batch_uploads_imageuploader                                                                                                                 
Use of uninitialized value $profile in concatenation (.) or string at ./batch_uploads_imageuploader line 33.
DBI connect('database=;host=;port=3306;','',...) failed: Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2) at /data/ibis/bin/mri/uploadNeuroDB/NeuroDB/DBI.pm line 52.
DB connection failed
DBI Error: Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)

```

May be something else is missing...

```
./batch_uploads_imageuploader -profile prod                                                                                                     
Possible unintended interpolation of @gmail in string at /data/ibis/bin/mri/dicom-archive/.loris_mri/prod line 36.
Use of uninitialized value $_ in pattern match (m//) at /data/ibis/bin/mri/batch_uploads_imageuploader line 50.
```

**Comment by MounaSafiHarab** [comment]
@gluneau 

Added the help section. 
You need to give this file a stdin input (text file with one entry per line for a scan to be inserted).
The help section now tells you all this (it was initially documented in: #133)

**Comment by MounaSafiHarab** [comment]
@gluneau 

Usage sections prints now without the -help

File: batch_uploads_imageuploader [hunk 1]
```diff
@@ -38,6 +62,12 @@ if ( $profile && !@Settings::db ) {
     exit 2;
 }
 
+if ( !$ARGV[0] || !$profile ) {
```
Comment (gluneau) [inline batch_uploads_imageuploader]: !$ARGV[0] || is not required

**Comment by gluneau** [comment]
Setting $profile = ''; instead of $profile = undef will prevent this error when -profile is not used:
Use of uninitialized value $profile in concatenation (.) or string at ./batch_uploads_imageuploader line 57.

**Comment by MounaSafiHarab** [comment]
@gluneau 

your requested changes are done. Thanks!

**Comment by gluneau** [comment]
Tested with:

```
lorisadmin@greg:/data/ibis/bin/mri% more list_of_scans.txt
/data/not_backed_up/ibis_fix_dcm/UNC1041_692620_V12_HOS2.zip N UNC1041_692620_V12
/data/not_backed_up/ibis_fix_dcm/UNC1047_837004_V06_HOS2.zip N UNC1047_837004_V06
```

File: batch_uploads_imageuploader [hunk 2]
```diff
@@ -111,10 +141,10 @@ foreach my $input (@resultsarray)
 	exit(4);
     }
     if ($phantom eq 'N') {
-        if ($base ne $patientname) {
+        if ($patientname ne (substr ($base, 0, length($patientname)))) {
```
Comment (MounaSafiHarab) [inline batch_uploads_imageuploader]: better to use regular expression

**Pull request merged**